server/bleep/src/webserver/answer.rs

```diff
@@ -292,18 +299,28 @@ impl AnswerAPIClient {
             #####
 
             Above is a code snippet.\
-            Your job is to answer the questions about the snippet,\
-            giving detailed explanations. Cite any code used to\
-            answer the question formatted in GitHub markdown and state the file path.
-
-            Q:What icon do we use to clear search history?
-            A:We use the left-double chevron icon.
+            Your job is to respond as a friendly customer support agent,\
+            answering the user's question with a detailed response.\
+            Separate each function out and explain why it is relevant.\
+            Format your response in GitHub markdown with code blocks annotated\
+            with programming language. Include the path of the file.
 
             Q:{}
             A:",
             snippet.relative_path, snippet.text, self.query
         );
-
-        self.send(prompt).await
+        let tokens_used = token_count_estimate(&prompt);
+        let max_tokens = 4097 - tokens_used;
+        self.send(prompt, max_tokens).await
     }
 }
+
+// openai claims 4 characters = 1 token for english,
+// but code typically clobbers the token count,
+// we get no more than 3 characters per token.
+//
+// TODO: this is a heuristic, a realistic tokenizer
+// can tune this better, rust_tokenizers::Gpt2Tokenizer?
+fn token_count_estimate(input: &str) -> u32 {
```

Comment:
Yes, the pretrained `GPT2Tokenizer` should be a better estimate. Could quickly run a side-by-side to see whether token estimates correlate: https://huggingface.co/docs/transformers/model_doc/openai-gpt

Reply:
will leave this as a TODO for now, and create a new ticket to tackle this.

Reply:
ill extract the tokenizer config from gpt2, and load that into our app for token count predictions.